pkg/asset/machines/aws/worker.go

```diff
@@ -7,12 +7,19 @@ import (
 	"github.com/openshift/installer/pkg/types"
 )
 
-// WorkerConfig is used to generate the worker machineset.
+// WorkerConfig is used to generate the worker machinesets.
 type WorkerConfig struct {
-	Replicas int64
+	Instances []WorkerMachinesetInstance
 	MachineConfig
 }
 
+// WorkerMachinesetInstance constains information specific to
+// one worker machineset.
+type WorkerMachinesetInstance struct {
+	Replicas         int64
+	AvailabilityZone string
```

Comment:
I'd rather add `Replicas` and `Zone` properties to `MachineConfig` (they're common to workers and masters), and maybe rename that to `MachineSetConfig`.  Then we could have:

```go
type WorkerConfig struct {
    MachineSets []MachineSetConfig
}
```

`MasterConfig` would be the same, except with a different name for our type-name-based asset handling.

Reply:
masters are `Machine` objects they don't have replicas. And all the fields in `MachineConfig` remain constant so doing that is replicating them for no reason.

Reply:
> masters are `Machine` objects they don't have replicas.

They're all going to end up as machine-sets though, right?  Why wouldn't master machine-sets need replica counts?

> And all the fields in `MachineConfig` remain constant..

That's not a requirement, is it?  You could have heterogenous nodes that differ by more than the zone.  Most folks will only need to change the zone, but a Go type that maps more directly to the machine-set object seems simpler and more flexible.

Reply:
Each master machine is currently unique. So it is modeled with Machine object. So if you ask installer for 3 masters it will give you 3 machine objects each in separate zone.

But as worker machines are cattle, they are modeled as Machineset object. 

So if you ask for 3 workers, the installer always creates almost identical Machineset for each az and stripes the worker count across them.

So each all non master pool is modeled as sets of machinsets for AWS..

We only do homogeneous pool, any change requires a separate pool.